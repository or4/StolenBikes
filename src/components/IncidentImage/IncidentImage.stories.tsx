import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, text } from '@storybook/addon-knobs';

import { IncidentImage } from './IncidentImage';

const stories = storiesOf('IncidentImage', module);
stories.addDecorator(withKnobs);

stories.add('IncidentImage', () => {
    return <IncidentImage imageUrl={getSrcKnobs()} />;
});

function getSrcKnobs() {
    return text(
        'Source',
        'https://avatars.mds.yandex.net/get-turbo/1505928/2a0000016836daee3ee7d60c44cca211b89b/max_g480_c12_r3x4_pd20',
    );
}
