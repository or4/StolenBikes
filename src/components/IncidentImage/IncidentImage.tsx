import React from 'react';
import styled from 'styled-components';

import prevewSvg from 'components/assets/preview.svg';
import { media } from 'ui/media';

interface IContainerProps {
    isPreview: boolean;
}

const Container = styled.div`
    ${media.mobile`
		display: ${(props: IContainerProps) => (props.isPreview ? 'none' : 'inherit')};
		padding-bottom: 10px;
  	`}

    ${media.tabletNDesktop`
		width: 100px;
		height: 100px;
		min-width: 100px;
		min-height: 100px;

		display: flex;
		align-items: center;
		justify-content: center;
		box-sizing: border-box;
		border-radius: 4px;
		border: 1px solid rgba(0, 0, 0, 0.05);
    	background: rgba(0, 0, 0, 0.02);
	`}
`;

interface Props {
    imageUrl?: string | null;
}

const Image = styled.div`
    background-image: url(${(props: Props) => props.imageUrl});
    background-repeat: no-repeat;
    background-size: cover;

    ${media.mobile`
		padding-top: 100%;
		width: 100%;
  	`}

    ${media.tabletNDesktop`
		width: 90px;
		height: 90px;
	`}
`;

const Preview = styled.div`
    background-image: url(${prevewSvg});
    background-repeat: no-repeat;
    background-size: cover;

    width: 55px;
    height: 32px;

    ${media.mobile`
		display: none;
  	`}
`;

export const IncidentImage = ({ imageUrl }: Props) => {
    // @ts-ignore
    const image = <Image imageUrl={imageUrl} />;

    // @ts-ignore
    return <Container isPreview={!Boolean(imageUrl)}>{imageUrl ? image : <Preview />}</Container>;
};
