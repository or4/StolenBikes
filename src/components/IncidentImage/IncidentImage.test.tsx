import React from 'react';
import { mount } from 'enzyme';

import { IncidentImage } from './IncidentImage';

describe('IncidentImage', () => {
    it('should render correctly', () => {
        const component = <IncidentImage />;

        const container = mount(component);
        expect(container).toMatchSnapshot();
        container.unmount();
    });
});
