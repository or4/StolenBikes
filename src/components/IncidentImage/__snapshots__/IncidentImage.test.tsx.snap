// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IncidentImage should render correctly 1`] = `
<IncidentImage>
  <styled.div
    isPreview={true}
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "eYDVYe",
            "rules": Array [
              "
    ",
              "
        @media ",
              "(max-width: ",
              "479",
              "px)",
              " {
            ",
              "
		display: ",
              [Function],
              ";
		padding-bottom: 10px;
  	",
              "
        }
    ",
              "

    ",
              "
        @media ",
              "(min-width: ",
              "480",
              "px)",
              " {
            ",
              "
		width: 100px;
		height: 100px;
		min-width: 100px;
		min-height: 100px;

		display: flex;
		align-items: center;
		justify-content: center;
		box-sizing: border-box;
		border-radius: 4px;
		border: 1px solid rgba(0, 0, 0, 0.05);
    	background: rgba(0, 0, 0, 0.02);
	",
              "
        }
    ",
              "
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      isPreview={true}
    >
      <div
        className="sc-bdVaJa eYDVYe"
      >
        <styled.div>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-htpNat",
                  "isStatic": true,
                  "lastClassName": "kYgIrP",
                  "rules": Array [
                    "
    background-image: url(",
                    "preview.svg",
                    ");
    background-repeat: no-repeat;
    background-size: cover;

    width: 55px;
    height: 32px;

    ",
                    "
        @media ",
                    "(max-width: ",
                    "479",
                    "px)",
                    " {
            ",
                    "
		display: none;
  	",
                    "
        }
    ",
                    "
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-htpNat",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="sc-htpNat kYgIrP"
            />
          </StyledComponent>
        </styled.div>
      </div>
    </StyledComponent>
  </styled.div>
</IncidentImage>
`;
